### DISCUSSION TITLE
Use RBAC Roles and RoleBindings instead of ClusterRoles

### DISCUSSION THREAD

**Discussion by kitos9112** [flant/shell-operator#410] [Q&A] [answered]
Hi!

I'm looking into using this operator in our production system, but I'm concerned about given super privs on granting a clusterRole for watching events (e.g., update secret) in a specific namespace. Is this something you're considering working on? or it's already available and I'm not able to make it work?

Thanks!

**Comment by diafour** [comment, marked as answer]
Hello! Examples are using ClusterRoles for simplicity or to watch for namespaces and cluster scoped CRDs. There is no requirement of ClusterRoles for shell-operator, you can define RoleBindings depending on bindings in your hooks.

Please attach more information to help you: hooks configuration, RBAC definitions, logs.

**Reply by kitos9112** [reply]
Hi @diafour 

Mea culpa, I had to explicitly specify the namespace selector in the Kubernetes config. I think an example with this would be beneficial as well. I'll see if I can open an PR covering https://github.com/flant/shell-operator/issues/131 and leveraging roles and roleBindings too.

```yaml
namespace:
  nameSelector:
    matchNames: ["${KUBERNETES_NAMESPACE}"]
```
